src/reqs.c

```diff
@@ -85,6 +85,97 @@
 #define CHECK_LWS(header, len)                                  \
   ((len) > 0 && (header[0] == ' ' || header[0] == '\t'))
 
+/*
+ * Read and parse the PROXY protocol v1 line, and update the client address.
+ */
+static int read_proxy_line (struct conn_s *connptr, union sockaddr_union* addr)
+{
+        int result = -1;
+        char *line = NULL;
+        ssize_t len;
+        char *src_addr, *end;
+        int is_ipv6;
+        int ret;
+
+        len = readline (connptr->client_fd, &line);
+        if (len <= 0) {
+                log_message (LOG_ERR,
+                             "read_proxy_line: Client (file descriptor: %d) "
+                             "closed socket before read.", connptr->client_fd);
+
+                goto cleanup;
+        }
+
+        /*
+         * Strip the new line and carriage return from the string.
+         */
+        if (chomp (line, len) == len) {
+                /*
+                 * If the number of characters removed is the same as the
+                 * length then it was a blank line.
+                 */
+                log_message (LOG_ERR, "Empty PROXY line found");
+
+                goto cleanup;
+        }
+
+        /*
+         * The PROXY line looks like that:
+         *   PROXY <proto> <src_addr> <dst_addr> <src_port> <dst_port>\r\n
+         * <proto> can be "TCP4" for IPv4/TCP or "TCP6" for IPv6/TCP
+         * <src_addr> and <dst_addr> are IPv4 or IPv6 addresses
+         * <src_port> and <dst_port> are TCP port numbers*
+         */
+
+        /* Expect "PROXY TCP4 " or "PROXY TCP6 " */
+        if (!(!strncmp (line, "PROXY TCP", 9)
+            && (line[9] == '4' || line[9] == '6')
+            && line[10] == ' ')) {
+                log_message (LOG_ERR, "Bad PROXY line: %s", line);
+
+                goto cleanup;
+        }
+
+        is_ipv6 = line[9] == '6' ? TRUE : FALSE;
```

Comment:
the use of the ternary here is redundant, `is_ipv6 = line[9] == '6';` is doing the same.

Reply:
actually, might as well make that `af = line[9] == '6' ? AF_INET6 : AF_INET;` and use af further down to fill in the address family and for the inet_pton call.